src/video-player.js

```diff
@@ -514,7 +556,8 @@ class VideoPlayer extends Utils.mixin(Eventable) {
   }
 
   source(publicId, options = {}) {
-    ({ publicId, options } = normalizeOptions(publicId, options));
+    const extendedOptions = Utils.assign({}, this.profileOptions.sourceOptions, options);
```

Comment:
Should not be done here, not all players go through here.
If you do this in the `cloudinary` plugin `source` method you can be sure it will be called on every souce-change